dbms/src/Functions/tests/gtest_json_array.cpp

```diff
@@ -33,7 +33,10 @@ class TestJsonArray : public DB::tests::FunctionTest
         static auto json_array_return_type = std::make_shared<DataTypeString>();
         assert(json_array_return_type->equals(*json_column.type));
         // The `json_binary` should be cast as a string to improve readability.
-        return executeFunction("cast_json_as_string", {json_column});
+        tipb::FieldType field_type;
+        field_type.set_flen(-1);
+        field_type.set_tp(TiDB::TypeString);
+        return executeCastJsonAsStringFunction({json_column}, field_type);
```

Comment:
Why not construct `filed_type` inside `executeCastJsonAsStringFunction`?

Reply:
FieldType has several attributes other than flen, like collations, thus pass it as a single parameter here.

Reply:
But for json type, the collation and flen should always be binary and -1?

Reply:
OK.